### PR TITLE
Update Hyperband Tutorial to new API

### PR DESCRIPTION
**Pull request by jdunnmon**
- [x] Updates `Tuning.ipynb` tutorial to new API
- [x] Recover tuning results from @agnusmaximus initial notebook

### CODE REVIEW

**Comment by ajratner** [comment]
@jdunnmon @agnusmaximus this is great, lmk when passing tests!  Initial comments:

- There's an argument for doing random search for the tutorial, and then just checking hyperband separately... but either way works
- Keep in mind that this is getting run by Travis, so please make sure it doesn't take too long to run (also note it's currently failing due to what looks like a `Queue` error- maybe run single threaded for tutorial?)
- For the tutorial, perhaps run in a less verbose mode?  Otherwise this generates an annoyingly large amount of diffs

Otherwise looks great so far, thanks!

**Comment by agnusmaximus** [comment]
Awesome, this PR makes the following changes:
- Updates Tuning.ipynb to use the new API
- Shortens tutorial (remove long hyperparameter search) to avoid timing out in the test (the Queue error)
- Add "verbose=False" to avoid too much output

**Comment by agnusmaximus** [comment]
Some more changes:
- Add a rng object to the Tuner to be used for config generator. This avoids the rng from being tangled with that used in model training.
- `search` call can optionally take a `seed` variable 
- _clear_state now resets the seed of rng object used in config generation

**Comment by agnusmaximus** [comment]
Some more changes:
- Add a rng object to the Tuner to be used for config generator. This avoids the rng from being tangled with that used in model training.
- `search` call can optionally take a `seed` variable 
- _clear_state optionally takes a seed to reset the rng to.

File: metal/tuners/tuner.py [hunk 1]
```diff
@@ -242,7 +247,7 @@ def search(
         raise NotImplementedError()
 
     @staticmethod
-    def config_generator(search_space, max_search, shuffle=True):
+    def config_generator(search_space, max_search, rng, shuffle=True):
```
[the remark this answers is not in the document]
Comment (bhancock8) [reply metal/tuners/tuner.py]: on second thought, I take it back. Looking more closely, I believe we made it a staticmethod to encourage us to keep the way configs are generated separated from the tuner that's checking different configs. I think that's still probably a good idea. So this is fine as is.

**Pull request merged**